horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/ActionServiceTest.java

```diff
@@ -60,4 +83,75 @@ public void testAddGlobalAction() {
       assertTrue(action.active);
       given().auth().oauth2(getAdminToken()).delete("/api/action/" + action.id);
    }
+
+   /**
+    * @param testInfo
+    */
+   @org.junit.jupiter.api.Test
+   public void testMessageBusAction(TestInfo testInfo) {
+      Test test = createTest(createExampleTest(getTestName(testInfo)));
+
+      String channel = "NOTACHANNEL";
+      String token = "BADTOKEN";
+
+      // Build an Action JSON directly because Action DTO serialization will
+      // mask the secrets. (Arguably, since we want to provoke an error here,
+      // the masked "****" value would work just as well, but this code runs
+      // successfully if the channel and token are real.)
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode config = mapper.createObjectNode()
+            .put("channel", channel).put("formatter", "testToSlack");
+      ObjectNode secrets = mapper.createObjectNode().put("token", token);
+      ObjectNode action = mapper.createObjectNode()
+            .put("id", -1)
+            .put("event", MessageBusChannels.TEST_NEW.name())
+            .put("type", SlackChannelMessageAction.TYPE_SLACK_MESSAGE)
+            .put("testId", test.id)
+            .put("active", true)
+            .put("runAlways", true);
+
+      // `set` returns a JsonNode, which can't be `set` or `put`, so we do
+      // these outside the chain.
+      action.set("config", config);
+      action.set("secrets", secrets);
+      Action r = jsonRequest().auth().oauth2(getAdminToken())
+            .contentType(ContentType.JSON).body(action)
+            .post("/api/action")
+            .then().statusCode(200)
+            .contentType(ContentType.JSON).extract().body().as(Action.class);
+
+      // send new test to action service
+      actionService.onNewTest(test);
+
+      log.infof("Waiting for action to run");
+      int errors = 0;
+      for (int i = 0; i < 10; i++) {
+         try {
+            Thread.sleep(1000);
+         } catch (InterruptedException e) {
+         }
+
+         // Should these action errors be logged under dummyTest.id? The
+         // ActionServiceImpl handler seems to always log under its argument
+         // value, -1: but I still can't find them.
+         List<ActionLog> logs = jsonRequest().auth().oauth2(getTesterToken())
```

Comment:
We use Row Level Security within the database to separate tenants. The action log messages are being written, but the security check is not returning the log messages from the rest API, I have opened an issue to track this: https://github.com/Hyperfoil/Horreum/issues/1353